### PR TITLE
Use multi-stage builds

### PR DESCRIPTION
**Pull request by eessex**
TODO:
- [ ] Ensure staging push builds properly
- [ ] Bump remote-docker-orb: https://github.com/artsy/orbs/pull/110

Establish workflow for muilti-stage docker builds in dev, test, and CI setup.
 
[WIP] https://artsyproduct.atlassian.net/browse/PLATFORM-2670

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -31,14 +41,28 @@ workflows:
...
           project_id: 38
 
-      - hokusai/test:
+      - artsy-remote-docker/build:
+          <<: *not_staging_or_release
+          context: hokusai
+          name: build
+          pre-steps:
+            - run:
+                command: echo 'export BUILD_TARGET="builder"' >> $BASH_ENV
```
Comment (izakp) [inline .circleci/config.yml:50]: Cool so this will persist the `BUILD_TARGET` env var between steps?

File: hokusai/build.yml [hunk 2]
```diff
@@ -1,9 +1,10 @@
 ---
-version: "2"
+version: "3.8"
 services:
   horizon:
+    image: "hokusai_horizon:${BUILD_TAG:-latest}"
     build:
-      context: ../
+      target: "${BUILD_TARGET}"
```
Comment (icirellik) [inline hokusai/build.yml:7]: This triggers a warning, still it works perfectly. In the future we may want to have a documented default target.

File: .circleci/config.yml [hunk 3]
```diff
@@ -3,6 +3,16 @@ version: 2.1
 orbs:
   hokusai: artsy/hokusai@0.7.5
   horizon: artsy/release@0.0.1
+  artsy-remote-docker: artsy/remote-docker@dev:25a57d9dad11f4ada9c4d9e42eb691f0
```
Comment (icirellik) [inline .circleci/config.yml:6]: Should be be set to the newly deploy orb version instead of a hash?

**Review by icirellik — APPROVED** [review]
Looks good aside from the orb version.

**Review by izakp — APPROVED** [review]
🎉

File: .circleci/config.yml (hunk 3 above)
Comment (izakp) [reply .circleci/config.yml]: Absolutely

**Pull request merged**